agimus_controller/agimus_controller/ocp/ocp_traj_tracking_collision_avoidance.py

```diff
@@ -0,0 +1,244 @@
+                    self._state, self._actuation, running_cost_model, constraints
+                )
+                running_model = crocoddyl.IntegratedActionModelEuler(running_DAM, dt)
+                running_model.differential.armature = self._robot_models.armature
+                running_model_list.append(running_model)
+        return running_model_list
+
+    def create_terminal_model(self) -> crocoddyl.ActionModelAbstract:
+        terminal_cost_model = crocoddyl.CostModelSum(self._state)
+        x_residual, _, frame_placement_residual = self.create_residuals()
+        x_reg_weights = np.ones(
+            self._robot_models.robot_model.nq + self._robot_models.robot_model.nv
+        )
+        x_reg_cost = crocoddyl.CostModelResidual(
+            self._state,
+            crocoddyl.ActivationModelWeightedQuad(x_reg_weights),
+            x_residual,
+        )
+        ee_pose_cost = crocoddyl.CostModelResidual(
+            self._state,
+            crocoddyl.ActivationModelWeightedQuad(np.ones(6)),
+            frame_placement_residual,
+        )
+
+        for col_idx in range(len(self._collision_model.collisionPairs)):
+            activation = ActivationModelQuadExp(
+                1, self._params.activation_distance_threshold**2
+            )
+            distance_coll_residual = ResidualDistanceCollision(
+                self._state, 7, self._collision_model, col_idx
+            )
+            distance_coll_cost = crocoddyl.CostModelResidual(
+                self._state, activation, distance_coll_residual
+            )
+            terminal_cost_model.addCost(f"distColl{col_idx}", distance_coll_cost, 1.0)
+        terminal_cost_model.addCost("stateReg", x_reg_cost, 1.0)
+        terminal_cost_model.addCost("goalTracking", ee_pose_cost, 1.0)
+        constraints = self.get_collision_constraints(active=True)
+        terminal_DAM = crocoddyl.DifferentialActionModelFreeFwdDynamics(
+            self._state, self._actuation, terminal_cost_model, constraints
+        )
+
+        terminal_model = crocoddyl.IntegratedActionModelEuler(terminal_DAM, 0.0)
+        terminal_model.differential.armature = self._robot_models.armature
+        return terminal_model
+
+    def get_collision_constraints(self, active=True):
+        """Returns the collision constraints model manager."""
+        constraint_model_manager = crocoddyl.ConstraintModelManager(
+            self._state, self.nq
+        )
+        if len(self._collision_model.collisionPairs) != 0:
+            for col_idx in range(len(self._collision_model.collisionPairs)):
+                collision_constraint = self._get_collision_constraint_residual(col_idx)
+                # Adding the constraint to the constraint manager
+                constraint_model_manager.addConstraint(
+                    "col_term_" + str(col_idx), collision_constraint, active=active
+                )
+        return constraint_model_manager
+
+    def _get_collision_constraint_residual(
```

Comment:
I do not understand the point of separating it into a separate function. This is short nought that it can be a part of `get_collision_constraints()`